File: app/src/main/java/com/example/seki/training_retrofit/ListAdapter.java
package com.example.seki.training_retrofit;

import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.seki.training_retrofit.databinding.ItemWeatherBinding;

import java.util.ArrayList;
import java.util.List;

public class ListAdapter extends RecyclerView.Adapter<ListAdapter.WeatherViewHolder>{

    final private List<WeatherData> mWeatherDatas;

    public ListAdapter() {
        mWeatherDatas = new ArrayList<>();
    }


    static class WeatherViewHolder extends RecyclerView.ViewHolder {
        final ItemWeatherBinding mBinding;
        WeatherViewHolder(View itemView) {
            super(itemView);
            mBinding = DataBindingUtil.bind(itemView);
        }
    }


    @Override
    public WeatherViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_weather, parent, false);
        return new WeatherViewHolder(view);
    }

    @Override
    public void onBindViewHolder(WeatherViewHolder holder, int position) {
        final WeatherData data = mWeatherDatas.get(position);
        holder.mBinding.setVariable(com.example.seki.training_retrofit.BR.data, data);
        holder.mBinding.executePendingBindings();
    }

    @Override
    public int getItemCount() {
        return mWeatherDatas.size();
    }

    public void addData(WeatherData data) {
        mWeatherDatas.add(0, data);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/seki/training_retrofit/ListAdapter.kt
package com.example.seki.training_retrofit;

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.seki.training_retrofit.databinding.ItemWeatherBinding

class ListAdapter: RecyclerView.Adapter<ListAdapter.WeatherViewHolder>(){

    private val mWeatherDatas: MutableList<WeatherData> = mutableListOf()


    class WeatherViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
        val mBinding: ItemWeatherBinding = DataBindingUtil.bind(itemView)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListAdapter.WeatherViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_weather, parent, false)
        return ListAdapter.WeatherViewHolder(view)
    }

    override fun onBindViewHolder(holder: ListAdapter.WeatherViewHolder, position: Int) {
        val data = mWeatherDatas[position]
        holder.mBinding.setVariable(com.example.seki.training_retrofit.BR.data, data)
        holder.mBinding.executePendingBindings()
    }

    override fun getItemCount() = mWeatherDatas.size

     fun addData(data: WeatherData) {
        mWeatherDatas.add(0, data)
        notifyDataSetChanged()
    }
}
